TELCODOCS-1052: Added requirements for installing an SNO cluster on AWS. (#57452)
openshift/openshift-docs: modules/install-sno_additional-requirements-for-installing-on-a-single-node-on-aws.adoc

// This module is included in the following assemblies: 
//
// installing/installing_sno/install-sno-preparing-to-install-sno.adoc

:_content-type: CONCEPT
[id="additional-requirements-for-installing-on-a-single-node-on-aws_{context}"]
= Additional requirements for installing on a single node on AWS

The AWS documentation for installer-provisioned installation is written with a high availability cluster consisting of three control plane nodes. When referring to the AWS documentation, consider the differences between the requirements for a {sno} cluster and a high availability cluster.

> * The required machines for cluster installation in AWS documentation indicates a temporary bootstrap machine, three control plane machines, and at least two compute machines. You require only one AWS instance for the control plane node and no worker nodes.

Reviewer: Hi @johnwilkins, `a temporary bootstrap machine` is still required by SNO.

Author: Fixed.

> * The minimum resource requirements for cluster installation in the AWS documentation indicates a control plane node with 4 vCPUs and 100GB of storage. For a single node cluster, you must have a minimum of 8 vCPU cores and 120GB of storage.

Reviewer: We have maintained a minimum resource table in [1], since they are the same, I suggest keeping [1] only, wdyt?

[1] https://57452--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_sno/install-sno-preparing-to-install-sno.html#install-sno-requirements-for-installing-on-a-single-node_install-sno-preparing

// This module is included in the following assemblies: 
//
// installing/installing_sno/install-sno-preparing-to-install-sno.adoc

:_content-type: CONCEPT

> [id="additional-requirements-for-installing-on-a-single-node-on-aws_{context}"]

Reviewer: Hello @johnwilkins, the current section structure is
```
Preparing to install on a single node
-- Additional requirements for installing on a single node on AWS  
    (Installing single-node OpenShift on AWS is mentioned in this section)

Installing OpenShift on a single node
-- Installing single-node OpenShift using the Assisted Installer
-- Installing single-node OpenShift manually
-- Creating a bootable ISO image on a USB drive
```
The _installation process_ is at the bottom of `Additional requirements ...` section, I think it's not clear enough to users, I would suggest to add a new section for AWS, e.g:
```
Preparing to install on a single node (un-touched)
-- Additional requirements for installing on a single node on AWS  
    (Installing single-node OpenShift on AWS is mentioned in this section: 
     Installing a single node cluster on AWS uses the "Installing a cluster on AWS with customizations" procedure.)

Installing OpenShift on a single node
-- Installing single-node OpenShift using the Assisted Installer
-- Installing single-node OpenShift manually
-- Installing single-node OpenShift on AWS
  ---- Requirements for installing on a single node on AWS
  ---- <Installation process>
-- Creating a bootable ISO image on a USB drive
```
WDYT？

Author: I created the new section for aws and put the additional requirements under there. I also created a module for installing with the process linked immediately after, since we do not incorporate xrefs into modules. I think you're right that it's clearer this way.

Author: @yunjiang29 add an lgtm if this is okay.

= Additional requirements for installing on a single node on AWS

The AWS documentation for installer-provisioned installation is written with a high availability cluster consisting of three control plane nodes. When referring to the AWS documentation, consider the differences between the requirements for an {sno} cluster and a high availability cluster.

> * The required machines for cluster installation in AWS documentation indicates a temporary bootstrap machine, three control plane machines, and at least two compute machines. You will require only one AWS instance for the control plane node and no worker nodes.

Reviewer 2: ```suggestion
* The required machines for cluster installation in AWS documentation indicates a temporary bootstrap machine, three control plane machines, and at least two compute machines. You require only one AWS instance for the control plane node and no worker nodes.
```

> * The minimum resource requirements for cluster installation in the AWS documentation indicates a control plane node with 4 vCPUs and 100GB of storage. For a single node cluster, you must have a minimum of 8 vCPU cores and 120GB of storage.

Reviewer: Hi @johnwilkins, how did you know 120GB is the minimum disk size?

> * The `controlPlane.replicas` setting in the `install-config.yaml` file should be set to `1`. This makes the control plane node schedulable.

Reviewer: ```suggestion
* The `controlPlane.replicas` setting in the `install-config.yaml` file should be set to `1`.
```

> * The `controlPlane.replicas` setting in the `install-config.yaml` file should be set to `1`. This will make the control plane node schedulable.

Reviewer 2: ```suggestion
* The `controlPlane.replicas` setting in the `install-config.yaml` file should be set to `1`. This makes the control plane node schedulable.
```

> * The `compute.replicas` setting in the `install-config.yaml` file should be set to `0`.

Reviewer: ```suggestion
* The `compute.replicas` setting in the `install-config.yaml` file should be set to `0`. This makes the control plane node schedulable.
```
